proposals/dictionary-expressions.md

```diff
@@ -55,44 +55,54 @@ Choices here would have implications regarding potential syntactic ambiguities,
 
 ## Design Intuition
 
-There are two core aspects to the design of dictionary expressions. 
+There are three core aspects to the design of dictionary expressions. 
 
-First is the concept of a *dictionary type*. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, with its `int`-to-`T` indexer. The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer. `Dictionary<TKey, TValue>` passes both requirements.
+1. Collection expressions containing `KeyValuePair<,>` (coming from `expression_element`, `spread_element`, or `key_value_pair_element` elements) can now instantiate a normal *collection type* *or* a *dictionary type*.
 
-Second is that collection expressions containing `KeyValuePair<,>` (coming from `expression_element`, `spread_element`, or `key_value_pair_element` elements) can now instantiate a normal *collection type* *or* a *dictionary type*.
+    So, if the target type for a collection expression is some *collection type* (that is *not* a *dictionary type*) with an element of `KeyValuePair<,>` then it can be instantiated like so:
 
-So, if the target type for a collection expression is some *collection type* (that is *not* a *dictionary type*) with an element of `KeyValuePair<,>` then it can be instantiated like so:
+    ```c#
+    List<KeyValuePair<string, int>> nameToAge = ["mads": 21];
+    ```
 
-```c#
-List<KeyValuePair<string, int>> nameToAge = ["mads": 21];
-```
+    This is just a simple augmentation on top of the existing collection expression rules.  In the above example, the code will be emitted as:
 
-This is just a simple augmentation on top of the existing collection expression rules.  In the above example, the code will be emitted as:
+    ```c#
+    __result.Add(new KeyValuePair<string, int>("mads", 21));
+    ```
 
-```c#
-__result.Add(new KeyValuePair<string, int>("mads", 21));
-```
+2. Introduction of the *dictionary type*. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, with its `int`-to-`T` indexer. The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer. `Dictionary<TKey, TValue>` passes both requirements.
```

Comment:
For `List<T>`, if it's the former requirement, should the reason be the following?

> The former requirement ensures that `List<T>` is not considered a dictionary type, since the *element type* is `T`.

Reply:
yup!